indigo/indigo/src/main/scala/indigo/gameengine/GameLoop.scala

```diff
@@ -47,101 +47,71 @@ class GameLoop[StartUpData, GameModel, ViewModel](
-                state._3
-            }
-
-          // Process events
-          scene.layers.foreach { layer =>
-            SceneGraphViewEvents.collectViewEvents(
-              boundaryLocator,
-              layer.nodes,
-              collectedEvents,
-              gameEngine.globalEventStream.pushGlobalEvent
-            )
-          }
-
-          // Play audio
-          gameEngine.audioPlayer.playAudio(scene.audio)
-
-          // Prepare scene
-          val sceneData = sceneProcessor.processScene(
-            gameTime,
-            scene,
-            gameEngine.assetMapping,
-            gameEngine.renderer.renderingTechnology,
-            gameConfig.advanced.batchSize
-          )
-
-          // Render scene
-          gameEngine.renderer.drawScene(sceneData, gameTime.running)
-        } else {
-          val processedFrame: Outcome[(GameModel, ViewModel)] =
-            frameProcessor.runSkipView(
-              startUpData,
-              gameModelState,
-              viewModelState,
-              gameTime,
-              collectedEvents,
-              inputState,
-              dice,
-              boundaryLocator
-            )
-
-          // Persist frame state
-          processedFrame match {
-            case oe @ Outcome.Error(e, _) =>
-              IndigoLogger.error("The game has crashed...")
-              IndigoLogger.error(oe.reportCrash)
-              throw e
-
-            case Outcome.Result(state, globalEvents) =>
-              gameModelState = state._1
-              viewModelState = state._2
-              globalEvents.foreach(e => gameEngine.globalEventStream.pushGlobalEvent(e))
-          }
+      // Persist frame state
+      val scene =
+        processedFrame match {
+          case oe @ Outcome.Error(e, _) =>
+            IndigoLogger.error("The game has crashed...")
+            IndigoLogger.error(oe.reportCrash)
+            throw e
+
+          case Outcome.Result(state, globalEvents) =>
+            gameModelState = state._1
+            viewModelState = state._2
+            globalEvents.foreach(e => gameEngine.globalEventStream.pushGlobalEvent(e))
+            state._3
         }
 
+      // Process events
+      scene.layers.foreach { layer =>
+        SceneGraphViewEvents.collectViewEvents(
+          boundaryLocator,
+          layer.nodes,
+          collectedEvents,
+          gameEngine.globalEventStream.pushGlobalEvent
+        )
       }
 
+      // Play audio
+      gameEngine.audioPlayer.playAudio(scene.audio)
+
+      // Prepare scene
+      val sceneData = sceneProcessor.processScene(
+        gameTime,
+        scene,
+        gameEngine.assetMapping,
+        gameEngine.renderer.renderingTechnology,
+        gameConfig.advanced.batchSize
+      )
+
+      // Render scene
+      gameEngine.renderer.drawScene(sceneData, gameTime.running)
+
+      // Tick
       gameEngine.platform.tick(gameEngine.gameLoop(time))
     } else gameEngine.platform.tick(loop(lastUpdateTime))
```

Comment:
Just a side note, it's a bit hard to follow the `if` for this `else` branch. I'd consider moving the whole `if` branch to a function 🤔